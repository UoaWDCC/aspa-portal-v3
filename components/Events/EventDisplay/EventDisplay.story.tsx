import { Meta, StoryObj } from "@storybook/react";
import { EventDisplay, displayType } from "./EventDisplay";
import type { EventCardMainProps } from "../EventCard/EventCardMain";
import type { EventCardSecondaryProps } from "../EventCard/EventCardSecondary";
import { Event } from "@/payload-types";

const eventsList: Event[] = [
  {
    id: 0,
    updatedAt: "2024-06-01T00:00:00.000Z",
    createdAt: "2024-05-01T00:00:00.000Z",
    name: "ASPA Hackathon @ GRID AKL",
    dateTime: "SAT & SUN 10:00-8:00pm",
    price: "10",
    location: "GridAKL [101 Pakenham St W]",
    description:
      "WDCC and SESA are so excited to announce our flagship hackathon for 2025. Join us for a two-day event where you alongside other students will work together to design and build unique applications - inspired by our surprise theme revealed on the day. Win from a prize pool of OVER $1.5k and grab some awesome sponsor merch along the way. Get mentored by experienced industry professionals - who will be on site ready to support your team!",
    imageUrl:
      "https://www.bartonmcgill.co.nz/wp-content/uploads/2023/12/51S4Ow-YSgL-_AC_SL1500_.jpg",
    date: null,
  },
  {
    id: 1,
    updatedAt: "2024-06-02T00:00:00.000Z",
    createdAt: "2024-05-02T00:00:00.000Z",
    name: "CASUAL NIGHT @ AKL CBD",
    dateTime: "TUES & THUR 6:30-8:00PM",
    price: "6",
    location: "Orange Pool Club [9 City Road]",
    description: "Join us for a casual night out at the Orange Pool Club!",
    imageUrl:
      "https://www.bartonmcgill.co.nz/wp-content/uploads/2023/12/51S4Ow-YSgL-_AC_SL1500_.jpg",
    date: null,
  },
  {
    id: 2,
    updatedAt: "2024-06-03T00:00:00.000Z",
    createdAt: "2024-05-03T00:00:00.000Z",
    name: "KARAOKE NIGHT @ AKL CBD",
    dateTime: "FRI 8:00-11:00PM",
    price: "10",
    location: "Orange Pool Club [9 City Road]",
    description: "Show off your singing skills at our Karaoke Night!",
    imageUrl:
      "https://gamesforfun.com/wp-content/uploads/2018/05/mini-ball-set-replacement-2-ball.jpg",
    date: null,
  },
];

const types = [displayType.Main, displayType.Secondary];

const propsArray = [
  // Main card props
  {
    bg_color: "#717882",
    font_color: "#FFFFFF",
    max_height: 350,
    min_height: 250,
    title_fs: "5vh",
    info_fs: "3vh",
    desc_fs: "2.5vh",
    text_margins: "sm",
    title_fw: "bolder",
    info_fw: "normal",
    desc_fw: "normal",
  } as EventCardMainProps,
  // Secondary card props
  {
    bg_color: "#717882",
    font_color: "#FFFFFF",
    max_height: 250,
    min_height: 200,
    title_fs: "3vh",
    info_fs: "2vh",
    text_gap: "lg",
    title_fw: "bold",
    info_fw: "normal",
  } as EventCardSecondaryProps,
];

const meta: Meta<typeof EventDisplay> = {
  title: "Events/EventDisplay",
  component: EventDisplay,
  argTypes: {
    events: { control: "object" },
    displayTypes: { control: "object" },
    props: { control: "object" },
    title: { control: "text" },
    buttonText: { control: "text" },
  },
};

export default meta;

export const Default: StoryObj<typeof EventDisplay> = {
  args: {
    events: eventsList,
    displayTypes: types,
    props: propsArray,
    title: "Upcoming Events",
    buttonText: "View More Events",
  },
};
