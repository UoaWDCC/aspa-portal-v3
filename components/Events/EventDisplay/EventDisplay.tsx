import { Event, EventCardMain } from "../EventCard/EventCardMain"
import { EventCardSecondary } from "../EventCard/EventCardSecondary"
import { Stack, Text, Button, rgba } from "@mantine/core"
import { useRouter } from "next/navigation"

export interface EventDisplayProps {
    events: Event[],
    title: string,
    buttonText: string,
    bg_color: string, 
    font_color: string,
    main_max_h: number,
    second_max_h: number,
    main_min_h: number,
    second_min_h: number,
}

export function EventDisplay({
    events,
    title,
    buttonText="View More Events",
    bg_color=rgba('#717882', 1), 
    font_color=rgba('#FFFFFF', 1), 
    main_max_h, 
    main_min_h, 
    second_max_h, 
    second_min_h
}: EventDisplayProps) {
    const router = useRouter();

    const handleViewMore = () => {
        router.push("/events");
    };

    return (
        <Stack align="center" gap="lg">
            <Text size="8vh" fw="bold">{title}</Text>
            
            <EventCardMain event={events[0]} bg_color={bg_color} font_color={font_color} max_height={main_max_h} min_height={main_min_h}/>

            <EventCardSecondary events={[events[1], events[2]]} bg_color={bg_color} font_color={font_color} max_height={second_max_h} min_height={second_min_h}/>

            <Button variant="filled" size="md" bg={bg_color} w="30vw" radius="xl" onClick={handleViewMore}>View More Events</Button>
        </Stack>
    )
}