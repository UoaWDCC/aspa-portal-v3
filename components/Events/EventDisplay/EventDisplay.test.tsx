import { render, screen } from "@/test-utils";
import { EventDisplay, displayType, EventCardProps } from "./EventDisplay";

const eventsList = [
  {
    id: "3",
    name: "ASPA Hackathon @ GRID AKL",
    date: "SAT & SUN 10:00-8:00pm",
    price: 10,
    location: "GridAKL [101 Pakenham St W]",
    description:
      "WDCC and SESA are so excited to announce our flagship hackathon for 2025. Join us for a two-day event where you alongside other students will work together to design and build unique applications - inspired by our surprise theme revealed on the day. Win from a prize pool of OVER $1.5k and grab some awesome sponsor merch along the way. Get mentored by experienced industry professionals - who will be on site ready to support your team!",
    imageUrl:
      "https://www.bartonmcgill.co.nz/wp-content/uploads/2023/12/51S4Ow-YSgL-_AC_SL1500_.jpg",
    buttonText: "Sign Up Now",
  },
  {
    id: "1",
    name: "CASUAL NIGHT @ AKL CBD",
    date: "TUES & THUR 6:30-8:00PM",
    price: 6,
    location: "Orange Pool Club [9 City Road]",
    description: "Join us for a casual night out at the Orange Pool Club!",
    imageUrl:
      "https://www.bartonmcgill.co.nz/wp-content/uploads/2023/12/51S4Ow-YSgL-_AC_SL1500_.jpg",
    buttonText: "See Details",
  },
  {
    id: "2",
    name: "KARAOKE NIGHT @ AKL CBD",
    date: "FRI 8:00-11:00PM",
    price: 11,
    location: "University Pool Club [15 Princes Street]",
    description: "Show off your singing skills at our Karaoke Night!",
    imageUrl:
      "https://gamesforfun.com/wp-content/uploads/2018/05/mini-ball-set-replacement-2-ball.jpg",
    buttonText: "See Details",
  },
];

const props: EventCardProps[] = [];

describe("EventDisplay component", () => {
  it("renders eventDisplay, eventCardMain, and eventCardSecondary correctly", () => {
    render(
      <EventDisplay
        events={eventsList}
        displayTypes={[displayType.Main, displayType.Secondary]}
        props={props}
      />,
    );
    expect(screen.getByText("Untitled")).toBeInTheDocument();
    eventsList.map((event, index) => {
      const img = screen.getByAltText(eventsList[index].name);
      expect(img).toBeInTheDocument();
      expect(img).toHaveAttribute("src", eventsList[index].imageUrl);

      expect(screen.getByText(eventsList[index].name)).toBeInTheDocument();
      expect(screen.getByText(eventsList[index].date)).toBeInTheDocument();
      expect(
        screen.getByText(`$${eventsList[index].price}`),
      ).toBeInTheDocument();
      expect(screen.getByText(eventsList[index].location)).toBeInTheDocument();
    });
  });

  it("renders eventCardTertiary correctly", () => {
    render(
      <EventDisplay
        events={[eventsList[0], eventsList[1]]}
        displayTypes={[displayType.Tertiary]}
        props={props}
      />,
    );
    expect(screen.getByText("Untitled")).toBeInTheDocument();
    [eventsList[0], eventsList[1]].map((event, index) => {
      const img = screen.getByAltText(eventsList[index].name);
      expect(img).toBeInTheDocument();
      expect(img).toHaveAttribute("src", eventsList[index].imageUrl);

      expect(screen.getByText(eventsList[index].name)).toBeInTheDocument();
      expect(screen.getByText(eventsList[index].date)).toBeInTheDocument();
      expect(
        screen.getByText(`$${eventsList[index].price}`),
      ).toBeInTheDocument();
      expect(screen.getByText(eventsList[index].location)).toBeInTheDocument();
    });
  });

  it("test for responsiveness", () => {
    const { rerender } = render(
      <EventDisplay
        events={eventsList}
        displayTypes={[displayType.Main, displayType.Secondary]}
        props={props}
      />,
    );

    expect(screen.getByText("Untitled")).toBeInTheDocument();

    global.innerWidth = 375;
    global.dispatchEvent(new Event("resize"));
    rerender(
      <EventDisplay
        events={eventsList}
        displayTypes={[displayType.Main, displayType.Secondary]}
        props={props}
      />,
    );

    expect(screen.getByText("Untitled")).toBeInTheDocument();
    eventsList.map((e, index) => {
      const img = screen.getByAltText(eventsList[index].name);
      expect(img).toBeInTheDocument();
      expect(img).toHaveAttribute("src", eventsList[index].imageUrl);

      expect(screen.getByText(eventsList[index].name)).toBeInTheDocument();
      expect(screen.getByText(eventsList[index].date)).toBeInTheDocument();
      expect(
        screen.getByText(`$${eventsList[index].price}`),
      ).toBeInTheDocument();
      expect(screen.getByText(eventsList[index].location)).toBeInTheDocument();
    });
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <EventDisplay
        events={eventsList}
        displayTypes={[displayType.Main, displayType.Secondary]}
        props={props}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
