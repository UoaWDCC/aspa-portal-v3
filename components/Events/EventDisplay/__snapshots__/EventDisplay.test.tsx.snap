// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDisplay component should match snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: fontFamily;--mantine-font-family-headings: fontFamily;--mantine-color-aspa-orange-0: #fff2e4;--mantine-color-aspa-orange-1: #fbe6d3;--mantine-color-aspa-orange-2: #f0ccaa;--mantine-color-aspa-orange-3: #e5af7e;--mantine-color-aspa-orange-4: #dc9758;--mantine-color-aspa-orange-5: #d78840;--mantine-color-aspa-orange-6: #d68032;--mantine-color-aspa-orange-7: #bd6d24;--mantine-color-aspa-orange-8: #a9611d;--mantine-color-aspa-orange-9: #945212;--mantine-color-aspa-brown-0: #faf5ef;--mantine-color-aspa-brown-1: #f0e8e0;--mantine-color-aspa-brown-2: #e2cdbb;--mantine-color-aspa-brown-3: #d5b192;--mantine-color-aspa-brown-4: #c9996f;--mantine-color-aspa-brown-5: #c38a59;--mantine-color-aspa-brown-6: #c1834d;--mantine-color-aspa-brown-7: #a96f3d;--mantine-color-aspa-brown-8: #986335;--mantine-color-aspa-brown-9: #845429;--mantine-color-aspa-gray-0: #f4f5f5;--mantine-color-aspa-gray-1: #e7e7e7;--mantine-color-aspa-gray-2: #cdcdcd;--mantine-color-aspa-gray-3: #b1b1b1;--mantine-color-aspa-gray-4: #999999;--mantine-color-aspa-gray-5: #8a8a8a;--mantine-color-aspa-gray-6: #838383;--mantine-color-aspa-gray-7: #707071;--mantine-color-aspa-gray-8: #636466;--mantine-color-aspa-gray-9: #52575b;}:root[data-mantine-color-scheme="dark"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-8);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-9);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.15);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.2);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-3);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-outline-hover: rgba(220, 151, 88, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-8);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-9);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.15);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.2);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-3);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-outline-hover: rgba(201, 153, 111, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-8);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-9);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.15);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.2);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-3);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-outline-hover: rgba(153, 153, 153, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-filled);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-7);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.1);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.12);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline-hover: rgba(214, 128, 50, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-filled);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-7);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.1);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.12);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline-hover: rgba(193, 131, 77, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-filled);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-7);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.1);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.12);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline-hover: rgba(131, 131, 131, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_6d731127 mantine-Stack-root"
    style="--stack-gap: var(--mantine-spacing-lg); --stack-align: center; --stack-justify: flex-start;"
  >
    <p
      class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      style="color: rgb(255, 255, 255); font-size: 8vh; font-weight: bold;"
    >
      Untitled
    </p>
    <a
      href="/events/3"
      style="text-decoration: none;"
    >
      <div
        class="m_8bffd616 mantine-Flex-root __m__-«r58»"
        style="position: relative; min-height: 250px; max-height: 350px; align-items: center; justify-content: flex-start; flex-direction: row; background: rgb(113, 120, 130);"
      >
        <img
          alt="ASPA Hackathon @ GRID AKL"
          class="m_9e117634 mantine-Image-root"
          src="https://www.bartonmcgill.co.nz/wp-content/uploads/2023/12/51S4Ow-YSgL-_AC_SL1500_.jpg"
          style="--image-object-fit: fill; width: auto; max-height: calc(21.875rem * var(--mantine-scale));"
        />
        <div
          class="m_8bffd616 mantine-Flex-root __m__-«r5b»"
          style="flex: 1; gap: var(--mantine-spacing-sm); justify-content: space-around; flex-direction: column; max-height: calc(21.875rem * var(--mantine-scale));"
        >
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            style="color: rgb(255, 255, 255); font-size: 5vh; font-weight: bolder; text-transform: uppercase;"
          >
            ASPA Hackathon @ GRID AKL
          </p>
          <div
            class="m_8bffd616 mantine-Flex-root __m__-«r5e»"
            style="gap: auto; align-items: center; justify-content: space-between; flex-direction: row; color: rgb(255, 255, 255);"
          >
            <p
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="font-size: 3vh; font-weight: normal;"
            >
              SAT & SUN 10:00-8:00pm
            </p>
            <p
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="font-size: 3vh; font-weight: normal;"
            >
              $10
            </p>
            <p
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="font-size: 3vh; font-weight: normal;"
            >
              GridAKL [101 Pakenham St W]
            </p>
          </div>
          <p
            class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
            data-line-clamp="true"
            style="--text-fz: 2.5vh; --text-lh: 2.5vh; --text-line-clamp: 4; color: rgb(255, 255, 255); font-weight: normal;"
          >
            WDCC and SESA are so excited to announce our flagship hackathon for 2025. Join us for a two-day event where you alongside other students will work together to design and build unique applications - inspired by our surprise theme revealed on the day. Win from a prize pool of OVER $1.5k and grab some awesome sponsor merch along the way. Get mentored by experienced industry professionals - who will be on site ready to support your team!
          </p>
          <div
            class="m_8bffd616 mantine-Flex-root __m__-«r5k»"
            style="align-items: flex-end; justify-content: flex-end; flex-direction: column;"
          >
            <button
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-size="compact-xl"
              data-variant="outline"
              style="--button-height: var(--button-height-compact-xl); --button-padding-x: var(--button-padding-x-compact-xl); --button-fz: var(--mantine-font-size-xl); --button-radius: var(--mantine-radius-xl); --button-bg: transparent; --button-hover: rgba(255, 255, 255, 0.05); --button-color: rgba(255, 255, 255, 1); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid rgba(255, 255, 255, 1); background: rgb(113, 120, 130);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  Sign Up Now
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </a>
    <div
      class="m_8bffd616 mantine-Flex-root __m__-«r5o»"
      style="min-height: 200px; max-height: 250px; gap: var(--mantine-spacing-md); align-items: center; justify-content: flex-start; flex-direction: row; width: 100%; flex: 1;"
    >
      <a
        href="/events/1"
      >
        <img
          alt="CASUAL NIGHT @ AKL CBD"
          class="m_9e117634 mantine-Image-root"
          src="https://www.bartonmcgill.co.nz/wp-content/uploads/2023/12/51S4Ow-YSgL-_AC_SL1500_.jpg"
          style="--image-object-fit: fill; width: auto; max-height: calc(15.625rem * var(--mantine-scale));"
        />
      </a>
      <a
        href="/events/2"
      >
        <img
          alt="KARAOKE NIGHT @ AKL CBD"
          class="m_9e117634 mantine-Image-root"
          src="https://gamesforfun.com/wp-content/uploads/2018/05/mini-ball-set-replacement-2-ball.jpg"
          style="--image-object-fit: fill; width: auto; max-height: calc(15.625rem * var(--mantine-scale));"
        />
      </a>
      <div
        class="m_8bffd616 mantine-Flex-root __m__-«r5s»"
        style="flex: 1; gap: var(--mantine-spacing-lg); justify-content: space-between; flex-direction: column;"
      >
        <a
          href="/events/1"
          style="text-decoration: none;"
        >
          <div
            class="m_6d731127 mantine-Stack-root"
            style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
          >
             
            <p
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="color: rgb(255, 255, 255); font-size: 3vh; font-weight: bold; text-transform: uppercase; width: 100%;"
            >
              CASUAL NIGHT @ AKL CBD
            </p>
            <div
              class="m_4081bf90 mantine-Group-root"
              style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap; color: rgb(255, 255, 255);"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-size: 2vh; font-weight: normal;"
              >
                TUES & THUR 6:30-8:00PM
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-size: 2vh; font-weight: normal;"
              >
                $6
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-size: 2vh; font-weight: normal;"
              >
                Orange Pool Club [9 City Road]
              </p>
            </div>
            <div
              class="m_8bffd616 mantine-Flex-root __m__-«r64»"
              style="align-items: flex-end; justify-content: flex-end; flex-direction: column;"
            >
              <button
                class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
                data-size="compact-md"
                data-variant="filled"
                style="--button-height: var(--button-height-compact-md); --button-padding-x: var(--button-padding-x-compact-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-xl); --button-bg: rgba(113, 120, 130, 1); --button-hover: rgba(102, 108, 117, 1); --button-color: var(--mantine-color-white); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; font-weight: normal; width: 15vw;"
                type="button"
              >
                <span
                  class="m_80f1301b mantine-Button-inner"
                >
                  <span
                    class="m_811560b9 mantine-Button-label"
                  >
                    See Details
                  </span>
                </span>
              </button>
            </div>
          </div>
        </a>
        <a
          href="/events/2"
          style="text-decoration: none;"
        >
          <div
            class="m_6d731127 mantine-Stack-root"
            style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
          >
             
            <p
              class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
              style="color: rgb(255, 255, 255); font-size: 3vh; font-weight: bold; text-transform: uppercase; width: 100%;"
            >
              KARAOKE NIGHT @ AKL CBD
            </p>
            <div
              class="m_4081bf90 mantine-Group-root"
              style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap; color: rgb(255, 255, 255);"
            >
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-size: 2vh; font-weight: normal;"
              >
                FRI 8:00-11:00PM
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-size: 2vh; font-weight: normal;"
              >
                $11
              </p>
              <p
                class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
                style="font-size: 2vh; font-weight: normal;"
              >
                University Pool Club [15 Princes Street]
              </p>
            </div>
            <div
              class="m_8bffd616 mantine-Flex-root __m__-«r6e»"
              style="align-items: flex-end; justify-content: flex-end; flex-direction: column;"
            >
              <button
                class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
                data-size="compact-md"
                data-variant="filled"
                style="--button-height: var(--button-height-compact-md); --button-padding-x: var(--button-padding-x-compact-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-xl); --button-bg: rgba(113, 120, 130, 1); --button-hover: rgba(102, 108, 117, 1); --button-color: var(--mantine-color-white); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; font-weight: normal; width: 15vw;"
                type="button"
              >
                <span
                  class="m_80f1301b mantine-Button-inner"
                >
                  <span
                    class="m_811560b9 mantine-Button-label"
                  >
                    See Details
                  </span>
                </span>
              </button>
            </div>
          </div>
        </a>
      </div>
    </div>
    <button
      class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
      data-size="md"
      data-variant="filled"
      style="--button-height: var(--button-height-md); --button-padding-x: var(--button-padding-x-md); --button-fz: var(--mantine-font-size-md); --button-radius: var(--mantine-radius-xl); --button-bg: var(--mantine-color-blue-filled); --button-hover: var(--mantine-color-blue-filled-hover); --button-color: var(--mantine-color-white); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; background: rgb(113, 120, 130); width: 30vw;"
      type="button"
    >
      <span
        class="m_80f1301b mantine-Button-inner"
      >
        <span
          class="m_811560b9 mantine-Button-label"
        >
          Button Text
        </span>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
